packages/manager/apps/telecom/src/app/telecom/telephony/alias/portability/portabilities/telecom-telephony-alias-portability-portabilities.controller.js

```diff
@@ -80,4 +83,17 @@ export default class TelecomTelephonyAliasPortabilitiesCtrl {
       this.loading.cancel = false;
     });
   }
+
+  loadDocuments(portability) {
+    const uploadedDocuments = [];
+    portability.documentAttached.forEach((documentId) => {
+      this.OvhApiTelephony.Portability().PortabilityDocument().v6().getDocument({
```

Comment:
sorry I don't understand what you want me to do. Could you explain it to me please ?

Reply:
Sorry about this :)

Right now you are doing this:
````
- set isLoading as true
- load all portabilities
- for each portability, launch a get to retrieve the associated document
- set isLoading as false
- still fetching some documents
````
which means that the datagrid is getting more and more documents gradually

what would be awesome is that you display a datagrid with all the documents already :)

It could be done by doing this instead:
````js
        if (results.documentAttached.length > 0) {
          return this.$q
            .all(results.documentAttached.map(documentId => this.OvhApiTelephony.Portability().PortabilityDocument().v6().getDocument({
              billingAccount: this.$stateParams.billingAccount,
              id: portability.id,
            }, {
              documentId,
            }).$promise))
            .then((documents) => {
              set(porta, 'uploadedDocuments', documents);
            });
        }
````

Do you understand the nuance ?

In the second code, your `Promise` won't be resolved until all documents have been retrieved, so you are sure you display a complete table to your users :). This makes paging, searching and filtering way easier

Reply:
Thanks, it's more clear for me now.